jlm/llvm/opt/alias-analyses/RegionAwareModRefSummarizer.hpp

```diff
@@ -95,84 +98,151 @@ class RegionAwareModRefSummarizer final : public ModRefSummarizer
 private:
   /**
    * Creates a call graph including all functions in the module, and groups all functions into SCCs.
-   * The resulting SCCs and topological order will be stored in the `FunctionSCCs_` field.
+   * The resulting SCCs and topological order will be stored in the Context_ field.
    *
    * @param rvsdgModule the module for which a mod/ref summary is computed.
    */
   void
   CreateCallGraph(const rvsdg::RvsdgModule & rvsdgModule);
 
   /**
-   * Creates RegionSummaries and CallSummaries for the given function.
-   * These summaries will contain the memory locations utilized in each region / call,
-   * including in sub-regions and inside call targets.
-   *
-   * Assumes that all functions with lower sccIndex have already been annotated.
-   * If regions or calls may contain recursion, a flag is set, to signal that the summaries are not
-   * complete. By the time all functions in an SCC have been annotated, the field
-   * SCCMemoryNodesAffected_ will contain the complete set of memory locations affected by the SCC.
+   * For each SCC in the call graph, determines which allocas can be known to not be live
+   * when a function from the SCC is at the top of the call stack.
    */
   void
-  AnnotateFunction(const rvsdg::LambdaNode & lambda, size_t sccIndex);
+  FindAllocasDeadInSccs();
 
   /**
-   * Recursive call used to create summaries for the given region and all sub-regions.
-   * @param region the region to create summaries for.
-   * @param sccIndex the SCC index of the function being analyzed
+   * Creates a set containing all simple Allocas is the PointsToGraph.
```

Comment:
allocas

Reply:
"IN the points-to graph" unless you want to reference the class, then use \ref please.